fvt/helpers.go

```diff
@@ -115,15 +139,15 @@ func UpdatePredictorAndWaitAndExpectFailed(predictorManifest *unstructured.Unstr
 
 	By("Waiting for the predictor " + predictorName + "'s target model state to move from Loaded (empty) to Loading")
 	loadingPredictor := WaitForLastStateInExpectedList("targetModelState", []string{"", "Loading"}, watcher)
-	ExpectPredictorState(loadingPredictor, true, "Loaded", "Loading", "InProgress")
+	ExpectState(loadingPredictor, true, "Loaded", "Loading", "InProgress")
 
 	By("Waiting for predictor " + predictorName + "'s target model state to be 'FailedToLoad'")
 	resultingPredictor := WaitForLastStateInExpectedList("targetModelState", []string{"", "Loading", "FailedToLoad"}, watcher)
-	ExpectPredictorState(resultingPredictor, true, "Loaded", "FailedToLoad", "BlockedByFailedLoad")
+	ExpectState(resultingPredictor, true, "Loaded", "FailedToLoad", "BlockedByFailedLoad")
 	return resultingPredictor
 }
 
-func ExpectPredictorState(obj *unstructured.Unstructured, available bool, activeModelState, targetModelState, transitionStatus string) {
+func ExpectState(obj *unstructured.Unstructured, available bool, activeModelState, targetModelState, transitionStatus string) {
```

Comment:
Hmm, not sure if status fields will be reflected in the object (in the case of InferenceServices) when this is being checked. Will need to check.

Reply:
Hi Paul, it is not reflected. I am unsure why this is so. If I try to get the object by `kubectl` then I can see the status field.

Reply:
Yea, I just tried attempting to reuse `WaitForLastStateInExpectedList` and it was giving me issues because sometimes a status field exists and sometimes it doesn't. Was thinking maybe its easier just to use something like this:

```golang
func WaitForIsvcReady(watcher watch.Interface) *unstructured.Unstructured {
	ch := watcher.ResultChan()
	isReady := false
	var obj *unstructured.Unstructured
	var isvcName string

	timeout := time.After(predictorTimeout)
	done := false
	for !done {
		select {
		// Exit the loop if InferenceService is not ready before given timeout.
		case <-timeout:
			done = true
		case event, ok := <-ch:
			if !ok {
				// the channel was closed (watcher timeout reached)
				done = true
				break
			}
			obj, ok = event.Object.(*unstructured.Unstructured)
			Expect(ok).To(BeTrue())
			isvcName = GetString(obj, "metadata", "name")
			conditions := GetSlice(obj, "status", "conditions")
			for _, condition := range conditions {
				conditionMap := condition.(map[string]interface{})
				if conditionMap["type"] == "Ready" {
					if conditionMap["status"] == "True" {
						isReady = true
						done = true
						break
					}
				}
			}

		}
	}
	Expect(isReady).To(BeTrue(), "Timeout before InferenceService '%s' ready", isvcName)
	return obj
}
```

Note: `GetSlice` is an added function in `utils.go` which is very similar to `GetString`. 

WDYT about just checking isvc status ready condition like the above?

Reply:
The object received at this stage looks like :
```
Obj: &unstructured.Unstructured{Object:map[string]interface {}{"apiVersion":"serving.kserve.io/v1beta1", "kind":"InferenceService", "metadata":map[string]interface {}{"annotations":map[string]interface {}{"serving.kserve.io/deploymentMode":"ModelMesh", "serving.kserve.io/secretKey":"localMinIO"}, "creationTimestamp":"2022-03-02T10:22:18Z", "generation":1, "managedFields":[]interface {}{map[string]interface {}{"apiVersion":"serving.kserve.io/v1beta1", "fieldsType":"FieldsV1", "fieldsV1":map[string]interface {}{"f:metadata":map[string]interface {}{"f:annotations":map[string]interface {}{".":map[string]interface {}{}, "f:serving.kserve.io/deploymentMode":map[string]interface {}{}, "f:serving.kserve.io/secretKey":map[string]interface {}{}}}, "f:spec":map[string]interface {}{".":map[string]interface {}{}, "f:predictor":map[string]interface {}{".":map[string]interface {}{}, "f:model":map[string]interface {}{".":map[string]interface {}{}, "f:modelFormat":map[string]interface {}{".":map[string]interface {}{}, "f:name":map[string]interface {}{}}, "f:storageUri":map[string]interface {}{}}}}}, "manager":"fvt.test", "operation":"Update", "time":"2022-03-02T10:22:18Z"}}, "name":"example-sklearn-isvc2-wz5g9", "namespace":"modelmesh-serving", "resourceVersion":"1549528", "uid":"48d6e243-ee52-40c0-ad04-a5899703b0df"}, "spec":map[string]interface {}{"predictor":map[string]interface {}{"model":map[string]interface {}{"modelFormat":map[string]interface {}{"name":"sklearn"}, "storageUri":"s3://modelmesh-example-models/sklearn/mnist-svm.joblib"}}}}}
```
It does not have field condition.

Reply:
Thank you, this works!